src/components/forms/SubmitTemplate/server-action.ts

```diff
@@ -0,0 +1,40 @@
+'use server';
+
+import { redirect, RedirectType } from 'next/navigation';
+import { getSession, saveTemplate } from '@utils/form-actions';
+import { createTemplateFromSession, validateTemplate } from '@domain/templates';
+import { logger } from '@utils/logger';
+
+export async function submitTemplate(formData: FormData) {
+  const sessionId = String(formData.get('sessionId'));
+
+  const session = await getSession(sessionId);
+
+  if (!session) {
+    return redirect('/invalid-session', RedirectType.push);
```

Comment:
Should be replace because we don't want to leave the bad URL in the history.  Also, I don't think you need to return, I think this throws (but please double check).

Reply:
updated

Reply:
Also updated to replace.